beacon-chain/cache/bloom_filter.go

```diff
@@ -0,0 +1,64 @@
+package cache
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/willf/bloom"
+)
+
+// ActiveIndicesBloomFilter defines BloomFilter for storing and twsting of active indices
+// Bloom Filter is a space-efficient probabilistic data structure
+// that is used to test whether an element is a member of a set
+type ActiveIndicesBloomFilter struct {
+	filter *bloom.BloomFilter
+	lock   sync.RWMutex
+}
+
+// A Bloom filter has two parameters: m - maximum size, k -the number of hashing functions on elements of the set
+const (
+	m = 300000
+	k = 5
+)
+
+// NewBloomFilter creates a new Bloom Filter for storing and testing active indices
+func NewBloomFilter() *ActiveIndicesBloomFilter {
+	return &ActiveIndicesBloomFilter{filter: bloom.New(m, k)}
+}
+
+// AddActiveIndicesBloomFilter adds a byte representation of active indices to a bloom filter
+func (bf *ActiveIndicesBloomFilter) AddActiveIndicesBloomFilter(byteIndices [][]byte) {
+	bf.lock.Lock()
```

Comment:
is a lock needed here ? does the bloom filter use a lock internally to protect against concurrent access ?

Reply:
I checked bloom filter source code https://github.com/willf/bloom, they dont use lock internally